packages/explat-client/src/internal/experiment-assignment-store.ts

```diff
@@ -4,15 +4,41 @@
 import type { ExperimentAssignment } from '../types';
 import * as Validations from './validations';
 
+const localStorageExperimentAssignmentsKey = 'explat-client-experiment-assignments';
```

Comment:
Instead of storing everything under a single key, why not use the experiment names as keys in the storage? We can add `explat_experiment_` as the key prefix to avoid collisions. This way we'd avoid a potentially-large JSON blob.

To deal with cases where localStorage isn't available, we can use a simple polyfill like [this one](https://gist.github.com/juliocesar/926500). We don't have to set it on `window`. Instead, we can set `ExperimentAssignmentStore.experimentNameToExperimentAssignment` directly to the storage instance: `window.localStorage` on a modern browser and an in-memory polyfill otherwise.